docs/tutorial/06-allow-snake-to-eat.md

```diff
@@ -0,0 +1,166 @@
+### 6. Allow snake to eat
+
+We have the food, but our snake can't eat it yet, and we don't want to starve our little friend.
+
+The goal of this chapter is to allow the snake to eat the food and grow, but let's first quickly recap how it works. The snake eats the food when it overlaps it with its head, as soon the snake eats it, it will grow of one unit (a new tile will be appended to its body) and a new food pellet will be placed in the snake world.
+
+In other words, we need to check if the snake's head has the same coordinates of the pellet after every movement. The function `move_snake/1` looks the natural place where to execute this check.
+
+```elixir
+defp move_snake(%{snake: snake} = state) do
+  %{body: body, size: size, direction: direction} = snake
+
+  # New head's position
+  [head | _] = body
+  new_head = move(state, head, direction)
+
+  # Place a new head on the tile that we want to move to
+  # and remove the last tile from the snake tail
+  new_body = List.delete_at([new_head | body], -1)
+
+  state
+  |> put_in([:objects, :snake, :body], new_body)
+  |> maybe_eat_food(new_head)
+end
+```
+
+👆Let's add it at the end of the state update pipeline, the pipe operator `|>` comes in handy here 💚
+
+Our new function `maybe_eat_food/2` receives:
+
+- The current state as 1st argument
+- The snake's head coordinates as 2nd argument
+
+Then, in the case the snake's head overlaps the food pellet, the function will take care of:
+
+- Grow the snake body of one unit
+- Place a new food pellet in the snake world
+
+or otherwise, just return the current state without any changes.
+
+Let's draft out the `maybe_eat_pellet/2` function and implement it step by step.
+
+```elixir
+def maybe_eat_pellet(state = %{pellet: pellet}, snake_head) do
+  if (pellet == snake_head)
+    state
+    |> grow_snake()
+    |> place_pellet()
+  else
+    state
+  end
+end
+```
+
+Growing the snake body can be tricky, let's explore all our alternatives.
+
+Prepending a new tile to the snake head is not a feasible solution because it will potentially lead to unexpected outcome like its dead :skull:.
+
+The most natural approach is to append a new tile at the end of the snake body, but how exactly? We can not simply add a new tile at the end like: `Enum.concat(body, [{x, y}])` since we don't have any information of the tail direction but only of its head. In other words, we can't infer the coordinates (`{x, y}`) of the tile to append. The only way to safely grow our snake it's to preserve its body when the snake ate the pellet. We could set a boolean flag `has_eaten` in the state and in the next game tick, don't delete the last snake's body tile when this flag is true 🤓.
```

Comment:
```suggestion
The most natural approach is appending a new tile to the end of the body. But how? We can't append a tile with `Enum.concat(body, [{x, y}])`. We can't infer the coordinates (`{x, y}`) for the new tile, since we don't know the tail's direction. We only know where the head is moving.

The only way to safely grow our snake its to preserve its body when it ate the pellet. We can set a boolean flag `has_eaten` in the state and on the next tick not delete the last body tile if is set to `true` 🤓. This will naturally grow our snake by one tile.
```